book/src/programs.md

```diff
@@ -1,32 +1,46 @@
 # Programming Model
 
-With the Solana runtime, we can execute on-chain programs concurrently, and
-written in the client’s choice of programming language.
+The Solana *runtime* can execute untrusted on-chain programs written in any
+programming language that can target the Berkley Packet Filter (BPF) safe
+execution environment. The Solana SDK offers the best support for C programs,
+which is compiled to BPF using the [LLVM compiler
+infrastructure](https://llvm.org). Alternatively, a client might choose to
+bypass LLVM and use Python, Lua or C++ to generate BPF directly via the [BPF
+Compiler Collection](https://github.com/iovisor/bcc) (BCC).
 
-## Client Interactions with Solana
+## Deploying Programs to a Cluster
 
 <img alt="SDK tools" src="img/sdk-tools.svg" class="center"/>
 
-As shown in the diagram above an untrusted client, creates a program in the
-language of their choice, (i.e. C/C++/Rust/Lua), and compiles it with LLVM to a
-position independent shared object ELF, targeting BPF bytecode, and sends it to
-the Solana cluster. Next, the client sends messages to the Solana cluster,
-which target that program. The Solana runtime loads the previously submitted
-ELF and passes it the client's message for interpretation.
+As shown in the diagram above a client creates a program and compiles it to an
+ELF shared object containing BPF bytecode and sends it to the Solana cluster.
+The cluster stores the program locally and makes it available to clients via a
+*program ID*. The program ID is a public key generated by the client and is
+used to reference the program in subsequent transactions.
 
-## Persistent Storage
+## Interacting with On-chain Programs
 
-Solana supports several kinds of persistent storage, called *accounts*:
+After a program has been deployed, a client utilizes it by sending transactions
+to the Solana cluster. A transaction contains one or more *instructions*. Each
+instruction references the program ID that coresponds to the program that the
+runtime should use to execute it. Instructions are executed atomically. If any
+instruction is invalid, any changes made within the transaction are discarded.
+
+## Storing State between Transactions
+
+If the the program needs to store state between transactions, it does so using
+*accounts*. Solana supports several kinds of accounts:
 
 1. Executable
 2. Owned by a client
 3. Owned by a program
 4. Credit-only
```

Comment:
I dont recognize this account type either from the code

Reply:
The code doesn't use the term, but that's how it's implemented. Credits to the "tokens" field are permitted, even if owned by another program.

Reply:
I wouldn't call that a different "kind" of account though.  One can apply credits to account kinds 1, 2, and 3 too, so all accounts are "Credit-only" (even executables accounts if you want to lose your tokens!)